Alec - Added Index Page for Students

alecsong222 opened the pull request:
## Overview
<!--A paragraph of the PR and related content-->
This PR adds the index page frontend functionality for Students, that shows the table, all students, and the edit, delete, and create buttons.
https://happycows-dev-alecsong222-students.dokku-09.cs.ucsb.edu/
https://6736df13168fb974aceba3b7-dtnvecyudi.chromatic.com/
Branch - https://github.com/ucsb-cs156-f24/proj-happycows-f24-09/tree/Alec-StudentsIndex
## Screenshots (Optional)
<!--Necessary screenshots and any necessary captions here. Delete if not needed.-->
![image](https://github.com/user-attachments/assets/11224ec1-51f4-4688-af6e-09cf6bfba4b7)

## Validation (Optional)
<!--Steps that someone else could take to make sure everything is working-->
Go to https://happycows-dev-alecsong222-students.dokku-09.cs.ucsb.edu/
Click on the Admin dropdown and go to Students
You should be taken to the Index page by default.
Ensure that the table shows up

## Tests
<!--Add any additional tests or required tests-->
- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues
<!--Issues related to the PR-->
Closes #48 

## PR Checklist
- [x] Commented code removed
- [x] PR deals with only one concern
- [x] PR Title is descriptive enough
- [x] PR description details what a user/dev will notice is different
- [x] No merge conflicts
- [x] Deployed and linked with Dokku deployment
- [x] New branch updated with main
- [x] PR is linked to a branch
- [x] PR is linked to an issue/ticket
- [x] Linked issue is in "In Review" column on Kanban board
- [x] PR is tagged with the Team's tag
- [x] All test cases pass

elijahelephant commented:
Merge main into your branch and then we should be good to add ready for staff.

pconrad commented on frontend/build-storybook.log:
```diff
@@ -0,0 +1,4 @@
+
+> frontend@0.1.0 build-storybook
+> build-storybook --docs -s public --output-dir=/var/folders/3b/x5mt4s9x3dx3ks913y481q340000gn/T/chromatic--65864-8YwseKd2A56g
+
```
Please remove from PR, and consider adding to `.gitignore`.

pconrad reviewed (changes requested):
One item to remove, and also wondering why the page isn't included in App.js and AppNavbar.js?

Unless/until it's included in App.js it can't actually be used.   We sometimes merge Form and Table components separately in order to increase the rate at which things can get merged, but Page components typically, we don't merge unless and until they are reachable by an end user.

pconrad reviewed (changes requested):
https://github.com/ucsb-cs156-f24/proj-happycows-f24-09/pull/43/files should not have been merged without adding validation for the courseId.

Something like this:
```
        coursesRepository.findById(courseId).orElseThrow(() -> new EntityNotFoundException(Courses.class, courseId));
```

I won't revoke the points you already earned.  But I also won't merge anything else that touches CRUD operations for Student (front end or backend) unless until this fix is part of that PR.

pconrad commented:
I appreciate the extra work on this PR

But there are still some problems that make this NOT Ready-For-Staff

There are two different dokku links in the PR description. Which one should I test with? Please clean this up.

FWIW: I tried testing with both of them, and neither of them was consistent with the recent commits on the branch for this PR.  

You can verify this by going to the SystemInfo endpoint on Swagger, and looking at the commit hash, and comparing with what's on this page: https://github.com/ucsb-cs156-f24/proj-happycows-f24-09/pull/54/commits

I'm going to remove Ready-For-Staff, and replace it with FIXME-Deploy.

* Please update the PR description so that there is only one dokku link
* Make sure that that dokku link has an up-to-date commit from your branch—should be the last commit (or nearly the last commit, modulo any merges to make the branch up-to-date with main that might happen after you deploy.)
* Then remove FIXME-Deploy and replace with Ready-For-Staff

alecsong222 commented:
Updated Dokku link with most recent deployment and removed extra dokku link. Please use https://happycows-dev-alecsong222-students.dokku-09.cs.ucsb.edu

pconrad reviewed (approved):
LGTM!
